test/xspec-xmlns-imported.xspec

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description xmlns=""
+  xmlns:q1="http://example.org/ns/my/ns1"
+  xmlns:tv1="http://example.org/ns/my/usercontent"
+  xmlns:x="http://www.jenitennison.com/xslt/xspec"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  stylesheet="xspec-xmlns.xsl">
+
+  <x:scenario label="Imported file using different declarations on x:description">
+    <x:call function="q1:param-mirror-fcn">
```

Comment:
When I insert two `x:variable` lines like this:
```xml
  <x:variable name="q1:global-var" select="'test'" />

  <x:scenario label="Imported file using different declarations on x:description">
    <x:variable name="q1:local-var" select="'test'" />
    <x:call function="q1:param-mirror-fcn">
```

Compiling `xspec-xmlns.xspec` fails:

```console
C:\xspec\test>java -cp "%SAXON_CP%" net.sf.saxon.Version
SAXON-EE 9.9.1.5J from Saxonica (build 090514)

C:\xspec\test>..\bin\xspec.bat xspec-xmlns.xspec
Saxon script not found, invoking JVM directly instead.

Creating Test Stylesheet...

Error evaluating (fn:resolve-QName(...)) in xsl:when/@test on line 719 column 37 of generate-common-tests.xsl:
  FONS0004: Namespace prefix 'q1' has not been declared
at template x:detect-reserved-variable-name on line 704 of generate-common-tests.xsl:
```

I don't know the root cause. For some reason, the error does not happen on Saxon 9.7.

When I replaced this `xsl:element`
https://github.com/xspec/xspec/blob/21ed0e1f844324b13670d7ad7828bb9894d2a26f/src/compiler/generate-common-tests.xsl#L161-L165

with `xsl:copy`, the compiler error went away somehow.

Reply:
Great catch! I will look into this further.

An alternate change that makes the compiler error go away is inserting `<xsl:sequence select="x:copy-namespaces(.)"/>`  before line 163 in `generate-common-tests.xsl`. It's likely that I had that instruction there at some point during my work but removed it because the lack of effect on any tests made me think it was unnecessary to copy namespaces at that spot. I'll also try other variations in the imported test file to see if other failures show up.

Reply:
In the code that handles imported files, namespace declarations are present in the output of the `match="node()|@*" mode="x:gather-specs"` template, just from the `xsl:copy` instruction. However, some namespace declarations seem to be missing when I examine the caller's `$all-specs` variable.

I'm aware of three alternate changes that make this error go away, at least for the case you found. I don't understand why they fix the problem, though.

A) Explicitly copy namespaces in the `match="node()|@*" mode="x:gather-specs"` template.

B) Explicitly copy namespaces in the `match="x:scenario" mode="x:gather-specs"` template.

C) Also in the `match="x:scenario" mode="x:gather-specs"` template, use `xsl:copy` instead of `xsl:element`.

Maybe there is a Saxon bug somewhere, given that 9.7 and 9.8 give different results. I spent a little time trying to reproduce the behavior in a small, self-contained XSLT file but don't have a conclusion on that.

In any case, 77d8571441d4a46906071d77f0a7769c65e0cb02 should fix this.

Reply:
While looking into this problem, I found a spot in `generate-xspec-tests.xsl` where namespaces need to be copied. I didn't think it was needed, but now I have a test case showing that it is (149773e84746a38e0f5dd92447bf0c40472716e0).

Reply:
Thanks for the fix.

> Maybe there is a Saxon bug somewhere, given that 9.7 and 9.8 give different results.

Looks like so. Namespace appears to be one of the most risky changes in every release.